pages/Navbar/Navbar.js

```diff
@@ -0,0 +1,109 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+export default function Navbar() {
+  const { data: session, status } = useSession();
+  const router = useRouter();
+
+  if (status === "loading") {
+    return <LoadingText>Loading...</LoadingText>;
+  }
+
+  return (
+    <NavbarContainer>
+      <UserSection>
+        {session ? (
+          <>
+            {session.user.image ? (
+              <Avatar src={session.user.image} alt={session.user.name} />
+            ) : (
+              <InitialAvatar>
+                {session.user.name.charAt(0)} {/* Display first letter */}
+              </InitialAvatar>
+            )}
+            <FullName>{session.user.name}</FullName>
+            <SignInButton onClick={() => signOut()}>Sign Out</SignInButton>
+          </>
+        ) : (
+          <SignInButton
+            onClick={() =>
+              signIn("credentials", { callbackUrl: router.asPath })
+            }
+          >
+            Sign In
+          </SignInButton>
+        )}
+      </UserSection>
+    </NavbarContainer>
+  );
+}
+
+const NavbarContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  background: linear-gradient(135deg, #ffdde1, #ee9ca7);
+  color: #333;
+  padding: 0 20px;
+  height: 60px;
+  box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.1);
+`;
+
+const UserSection = styled.div`
```

Comment:
If you rename the container, I would change that one as well:

```suggestion
const UserDetails = styled.div`
```